### ISSUE TITLE
Add a `golden-test new-test` feature

### ISSUE THREAD

**Issue by drevell** [abcxyz/abc#316]
### TL;DR

Currently, to add a new golden test, the user must go create a file named testdata/golden/$testname/test.yaml

This has some UX issues:
 - The user has to know what directory structure is required
 - The user has to write a YAML file by hand, and know the field names

It would be more convenient if the user could create new tests from the command line:
 - Option 1: we can add flags to the existing `golden-test record` subcommand to give it the ability to create a test.yaml file. Example: `abc templates golden-test record --location . --test-name my-new-test --input service_account=abc123@foo.com --input somethingelse=bar`
 - Option 2: we add a new subcommand `abc templates golden-test new-test`. Example: `abc templates golden-test new-test --location . --test-name my-new-test --input service_account=abc123@foo.com --input somethingelse=bar`

We could also support prompting for inputs: `abc templates golden-test new-test --location . --test-name my-new-test --prompt`

cc @chloechien

**Comment by capri-xiyue** [comment]
Hi @drevell,  I did a quick investigation, I prefer we use option 22 with a new subcommand `abc templates golden-test new-test` and at the same time we support prompting for inputs. I think option 2 is more clear to users as we separate the `new-test` and `record` feature and it satisfies the single responsibility principle. What do you think of it?

**Comment by drevell** [comment]
> Hi @drevell, I did a quick investigation, I prefer we use option 2 with a new subcommand `abc templates golden-test new-test` and at the same time we support prompting for inputs. I think option 2 is more clear to users as we separate the `new-test` and `record` feature and it satisfies the single responsibility principle. What do you think of it?

@capri-xiyue Makes sense, sounds good to me.

**Comment by drevell** [comment]
@capri-xiyue can we close this issue now? I think this might be finished.

**Comment by capri-xiyue** [comment]
> @capri-xiyue can we close this issue now? I think this might be finished.

closed
